dora/core/server/master/src/main/java/alluxio/master/job/RoundRobinWorkerAssignPolicy.java

```diff
@@ -13,23 +13,31 @@
 
 import alluxio.wire.WorkerInfo;
 
+import java.util.ArrayList;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
 import java.util.concurrent.atomic.AtomicInteger;
 
 /**
  * Policy which employs Round Robin around given workers to select worker.
  */
-public class RoundRobinWorkerAssignPolicy extends WorkerAssignPolicy {
+public class RoundRobinWorkerAssignPolicy implements WorkerAssignPolicy {
   private AtomicInteger mCounter = new AtomicInteger(0);
 
   @Override
-  protected WorkerInfo pickAWorker(String object, Collection<WorkerInfo> workerInfos) {
-    if (workerInfos.isEmpty()) {
-      return null;
+  public List<WorkerInfo> pickWorkers(String object, Collection<WorkerInfo> workerInfos,
+      int count) {
+    if (count > workerInfos.size()) {
+      return Collections.emptyList();
     }
-    int nextWorkerIdx = Math.floorMod(mCounter.incrementAndGet(), workerInfos.size());
-    WorkerInfo pickedWorker = workerInfos.toArray(new WorkerInfo[workerInfos.size()])
-        [nextWorkerIdx];
-    return pickedWorker;
+    ArrayList workers = new ArrayList<>();
+    for (int i = 0; i < count; i++) {
+      int nextWorkerIdx = Math.floorMod(mCounter.incrementAndGet(), workerInfos.size());
+      WorkerInfo pickedWorker =
+          workerInfos.toArray(new WorkerInfo[workerInfos.size()])[nextWorkerIdx];
+      workers.add(pickedWorker);
```

Comment:
do deduplication , i m assuming when we call pickWorkers we always wants to be pick different "count" of workers right? need to call it out in interface doc if thats the case

Reply:
In worker location policy there's dedup so no need to do it here

Reply:
no this is roundrobin

Reply:
oh, so if workerinfos are distinct workers then this round robin should return distinct workers I think? I would update the method signature

Reply:
oh i get what u r saying, u r right here.